src/features/comments/commentsSlice.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-param-reassign */
+
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Comment } from '../../types/Comment';
+import { getPostComments } from '../../api/comments';
+
+export const fetchComments = createAsyncThunk<Comment[], number>(
+  'comments/fetchComments',
+  postId => {
+    const response = getPostComments(postId);
+
+    return response;
+  },
+);
+
+const initialState: {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+} = {
+  comments: [],
+  loaded: true,
+  hasError: false,
+};
+
+export const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {
+    setComments: (state, action: PayloadAction<Comment>) => {
+      return {
+        ...state,
+        comments: [...state.comments, action.payload],
+      };
```

Comment:
The `setComments` reducer seems to be designed to add a single comment, but the payload is typed as `Comment` which represents a single comment. However, the action is named in plural which is misleading. Consider renaming the action to `addComment` to better reflect its purpose.